jmespath-core/src/main/java/io/burt/jmespath/BaseAdapter.java

```diff
@@ -60,8 +58,8 @@ public int compare(T value1, T value2) {
         case BOOLEAN:
           return (isTruthy(value1) && isTruthy(value2)) || (!isTruthy(value1) && !isTruthy(value2)) ? 0 : -1;
         case NUMBER:
-          Comparable d1 = toNumber(value1).doubleValue();
-          Comparable d2 = toNumber(value2).doubleValue();
+          Double d1 = toNumber(value1).doubleValue();
+          Double d2 = toNumber(value2).doubleValue();
           return d1.compareTo(d2);
```

Comment:
I don't think `boolean` is comparable in JMESPath:

> Ordering operators >, >=, <, <= are only valid for numbers. Evaluating any other type with a comparison operator will yield a null value, which will result in the element being excluded from the result list.

Or did you mean something else by "the documentation"? (reading that I think we're not compliant for another reason, I think we raise an error for non-numeric types).

True about the `Double`/`double` thing. There might be more of those.